spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubAutoConfiguration.java

```diff
@@ -118,11 +121,17 @@ public ExecutorProvider subscriberExecutorProvider() {
 				this.gcpPubSubProperties.getSubscriber().getExecutorThreads()));
 	}
 
+	@Bean
+	@ConditionalOnBean(type = "com.fasterxml.jackson.databind.ObjectMapper")
+	@ConditionalOnMissingBean
+	public PubSubMessageConverter pubSubMessageConverter(ObjectMapper objectMapper) {
+		return new JacksonPubSubMessageConverter(objectMapper);
+	}
+
 	@Bean
 	@ConditionalOnMissingBean
 	public PubSubMessageConverter pubSubMessageConverter() {
-		return new JacksonPubSubMessageConverter(
-				this.gcpPubSubProperties.getTrustedPackages());
+		return new SimplePubSubMessageConverter();
```

Comment:
Hm... Do we really need a bean for this variant?
I see it is a default instance in the target `PubSubTemplate`.
So, maybe we really can live without a default bean?
The `PubSubTemplate` bean has to be fixed for the `ObjectProvider<PubSubMessageConverter>` instead though...

Reply:
Sounds good.

Reply:
You missed to remove this bean.